src/tools/illink/src/linker/Linker.Steps/MarkStep.MarkStepNodeFactory.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using Mono.Cecil;
+
+namespace Mono.Linker.Steps
+{
+	public partial class MarkStep
+	{
+		internal sealed class MarkStepNodeFactory (MarkStep markStep)
```

Comment:
```suggestion
		internal sealed class NodeFactory (MarkStep markStep)
```

Nit: We just call this NodeFactory in both crossgen and ilc (it's a different class, just same name). We're going to type this a lot so a shorter name that doesn't have the same prefix as another class will be more convenient.